pipelined/bevy_sprite2/src/render/mod.rs

```diff
@@ -85,19 +86,51 @@ impl FromWorld for SpriteShaders {
             depth_stencil: None,
             vertex: VertexState {
                 buffers: &[VertexBufferLayout {
-                    array_stride: 20,
-                    step_mode: InputStepMode::Vertex,
+                    array_stride: mem::size_of::<SpriteInstance>() as BufferAddress,
+                    step_mode: InputStepMode::Instance,
                     attributes: &[
+                        // sprite_transform_0
                         VertexAttribute {
-                            format: VertexFormat::Float32x3,
+                            format: VertexFormat::Float32x4,
                             offset: 0,
                             shader_location: 0,
                         },
+                        // sprite_transform_1
                         VertexAttribute {
-                            format: VertexFormat::Float32x2,
-                            offset: 12,
+                            format: VertexFormat::Float32x4,
+                            offset: 16,
```

Comment:
Should we be using consts for the magic numbers here? It would likely be clearer to beginners if nothing else.

The values have already needed to change for this PR; hopefully a nice set of constants would make that process easier and more reliable.